src/libaktualizr/package_manager/CMakeLists.txt

```diff
@@ -58,7 +58,7 @@ aktualizr_source_file_checks(ostreemanager.cc ostreereposync.cc
                              ostreemanager.h ostreereposync.h)
 
 aktualizr_source_file_checks(androidmanager.cc androidmanager.h)
-aktualizr_source_file_checks(dockerappmanager.cc dockerappmanager.h dockerappmanager_test.cc)
+aktualizr_source_file_checks(dockerappmanager.cc dockerappmanager.h dockerapp_standalone.cc dockerapp_bundles.cc dockerappmanager_test.cc)
```

Comment:
This change https://github.com/advancedtelematic/aktualizr/commit/d06aa4fb06e408b03df39b1fe0216e3ef05d8e97 provides a capability to build the dockerapp pacman out the libaktualizr's cmakes/build system as well as register it without a need in libaktualizr code modification.

Reply:
I'm quite interested in the pluggable package managers. I could you all eventually wanting me to maintain this out of tree. For now, I'm hoping to keep in tree because of all the advantages of being in the upstream. However, I can understand if over time this just isn't worth the effort for you to worry about (ie merge conflict type stuff).

Reply:
The dockerapp package manager actually consists of two package managers, ostree and dockerapp. It potentially can be split into two package managers, but libaktualizr doesn't support multiple package managers. Another alternative here is to use the secondary mechanism to extend Primary's default ostree update with the dockeapp one. In this case, the dockerapp package manager will be represented as Secondary in libaktualizr, which allows you to build it and register at libaktualizr without modifying libaktualizr code.

Reply:
> I'm quite interested in the pluggable package managers. I could you all eventually wanting me to maintain this out of tree. For now, I'm hoping to keep in tree because of all the advantages of being in the upstream. However, I can understand if over time this just isn't worth the effort for you to worry about (ie merge conflict type stuff).

That's still an open topic of discussion. For now, I think it's fine, especially since it's pretty tightly coupled due to the lack of a proper TUF API. If we had something like that, though, and we got it to a point that it was reasonably stable, then maybe it would be time to separate it. But for now, it might even be beneficial to have it directly in this project so that we know what the expectations are for the use cases that make the DockerApp package manager unique.

Part of the point of the pluggable package manger, though, is to make the barrier of entry to people making their own lower.

Both of Mike's point are quite salient as well. Supporting multiple package managers is another long-term goal of ours, and again, we'll want to make sure that whatever solution we end up with works for your use case. Treating the DockerApps as independent Secondaries is logically closer to what we've recommended to other users in the meantime until we have proper support for multiple package managers (and targets) on one ECU, but I don't know if it's worth the effort for you to rework the existing system to fit that mold.

Reply:
I've got a TODO for understanding secondaries more deeply. One thing we like about the current approach is that an update is a single immutable Target. However, it comes at a cost. The other thing I'd have to figure out is migration path(s) for existing devices.